services/gateway/endpoints/src/main/java/org/eclipse/ditto/services/gateway/endpoints/directives/ContentTypeValidationDirective.java

```diff
@@ -0,0 +1,111 @@
+ /*
+  * Copyright (c) 2020 Contributors to the Eclipse Foundation
+  *
+  * See the NOTICE file(s) distributed with this work for additional
+  * information regarding copyright ownership.
+  *
+  * This program and the accompanying materials are made available under the
+  * terms of the Eclipse Public License 2.0 which is available at
+  * http://www.eclipse.org/legal/epl-2.0
+  *
+  * SPDX-License-Identifier: EPL-2.0
+  */
+ package org.eclipse.ditto.services.gateway.endpoints.directives;
+
+ import static akka.http.javadsl.server.Directives.complete;
+ import static org.eclipse.ditto.services.gateway.endpoints.utils.DirectivesLoggingUtils.enhanceLogWithCorrelationId;
+
+ import java.text.MessageFormat;
+ import java.util.Optional;
+ import java.util.Set;
+ import java.util.function.Supplier;
+ import java.util.stream.StreamSupport;
+
+ import org.eclipse.ditto.model.base.headers.DittoHeaderDefinition;
+ import org.eclipse.ditto.model.base.headers.DittoHeaders;
+ import org.slf4j.Logger;
+ import org.slf4j.LoggerFactory;
+
+ import akka.http.javadsl.model.HttpHeader;
+ import akka.http.javadsl.model.HttpRequest;
+ import akka.http.javadsl.model.StatusCodes;
+ import akka.http.javadsl.server.RequestContext;
+ import akka.http.javadsl.server.Route;
+
+ /**
+  * Used to validate the Content-Type of a request.
+  */
+ public final class ContentTypeValidationDirective {
+
+     private static final Logger LOGGER = LoggerFactory.getLogger(ContentTypeValidationDirective.class);
+
+     /**
+      * verifies that the content-type of the entity is one of the allowed media-types,
+      * otherwise the request will be completed with 415 ("Unsupported Media Type").
+      *
+      * @param supportedMediaTypes the media-type which are allowed for the wrapped route.
+      * @param ctx the context of the request.
+      * @param dittoHeaders the ditto-headers of a request.
+      * @param inner route to wrap.
+      * @return the wrapped route.
+      */
+     public static Route ensureValidContentType(final Set<String> supportedMediaTypes, final RequestContext ctx,
+             final DittoHeaders dittoHeaders,
+             final Supplier<Route> inner) {
+         return enhanceLogWithCorrelationId(dittoHeaders.getCorrelationId(), () -> {
+             final String requestsMediaType = extractMediaType(ctx.getRequest());
+
+             if(supportedMediaTypes.contains(requestsMediaType)){
```

Comment:
Does this also work with content-types which define the charset?
I did not see at first sight any logic handling the charset part in a Content-Type..

Reply:
The media-type is the content-type without the charset. That's how i omit the charset to avoid complications. Only in the case where an illegal/non-parsable content-type should be allowed, the configured whitelisted value has to match the raw-header equally with charset.